util/crc32c.cc

```diff
@@ -315,9 +315,16 @@ static inline void Slow_CRC32(uint64_t* l, uint8_t const **p) {
 
 static inline void Fast_CRC32(uint64_t* l, uint8_t const **p) {
 #ifdef __SSE4_2__
+#ifdef __LP64__
   *l = _mm_crc32_u64(*l, LE_LOAD64(*p));
   *p += 8;
 #else
+  *l = _mm_crc32_u32((unsigned int)*l, LE_LOAD32(*p));
+  *p += 4;
+  *l = _mm_crc32_u32((unsigned int)*l, LE_LOAD32(*p));
```

Comment:
Can you please use static_cast<unsigned int>(*l). We prefer C++ style casts to C style casts in RocksDB codebase.